### DISCUSSION TITLE
Auth by Apple ID

### DISCUSSION THREAD

**Discussion by elsacutealways** [pocketbase/pocketbase#2446] [Q&A]
How to implement apple authentication in flutter app for iOS ?

**Comment by ganigeorgiev** [comment]
The question is too vague and I'm not sure how to help.

If you haven't already, you could check https://pocketbase.io/docs/authentication/#oauth2-integration.

The Apple provider requires a client id and secret.
The client id is the Apple OAuth2 App Id.
The client secret can be generated from the PocketBase "Admin UI > Auth Providers > Apple" settings by specifying an App Id, Team Id, Key Id and the Private Key of your app (_it is not stored anywhere and it is used only for the secret generation_).

This medium article - https://medium.com/identity-beyond-borders/how-to-configure-sign-in-with-apple-77c61e336003 covers some of the aspects how to configure the OAuth2 app in the Apple developer dashboard (_you can skip the manual secret generation step since PocketBase comes with builtin generator_).

**Reply by lukepighetti** [reply]
any chance we could get a summary of how to implement SIWA with pocketbase on flutter? There are so many moving parts in the integration and it's a very opaque integration.

I thought I had followed all the instructions in the above links but I keep getting a "Sign up not completed" error in the SIWA native dialog and I don't see any logs in pocketbase showing activity on my redirect url.

**Reply by lukepighetti** [reply]
heres a guide for Appwrite that seems almost complete. What are we supposed to use for client id in Pocketbase token generation modal? 

https://dev.to/appwrite/apple-sign-in-with-appwrite-2576

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand what do you mean by "SIWA native dialog", but you can try to follow the instructions from https://github.com/pocketbase/pocketbase/discussions/5074.

I don't have Apple developer account to help more than this.

**Reply by lukepighetti** [reply]
do you have access to an iPhone / macOS environment?

**Reply by ganigeorgiev** [reply]
No, sorry. In the [above discussion](https://github.com/pocketbase/pocketbase/discussions/5074) you could find link to the Okta guide that contains screenshots which may help you with the required setup in the Apple dashboard.

_Keep in mind that the Apple OAuth2 integration was implemented based only on their docs and without me manually testing it. It is the only such PocketBase provider and if it wasn't for the Apple store requirements I wouldn't even bother adding it as I don't think I can call Apple an OAuth2 OIDC provider in the first place (although they advertise themselves as such) because there are way too many subtle differences and caveats for something that implements a standard._

**Reply by lukepighetti** [reply]
got it... thanks!
